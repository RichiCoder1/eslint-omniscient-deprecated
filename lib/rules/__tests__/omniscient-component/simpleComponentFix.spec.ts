import { fix, withRuleOptions } from "./__setup__/setup";

describe("simple component conversions", () => {
    it("should convert render only component", () => {
        const input = `
import component from 'omniscient';

const TestComponent = component(() => {
    return (<h1>Test</h1>);
})`;
        const output = fix(input);
        expect(output).toBeFixed();
        expect(output).toMatchSnapshot();
    });

    it("should convert simple pure component", () => {
        const input = `
import component from 'omniscient';

const TestComponent = component("TestComponent", () => {
    return (<h1>Test</h1>);
})`;
        const output = fix(input);
        expect(output).toBeFixed();
        expect(output).toMatchSnapshot();
    });

    it("should convert simple pure component to function when has memo", () => {
        const input = `
import component from 'omniscient';

const TestComponent = component("TestComponent", () => {
    return (<h1>Test</h1>);
})`;
        const output = fix(
            input,
            withRuleOptions({
                memoImport: "memo",
                memoModule: "react",
                passAreEqualToMemo: false,
            })
        );
        expect(output).toBeFixed();
        expect(output).toMatchSnapshot();
    });

    it("should convert simple component to function when has memo", () => {
        const input = `
import component from 'omniscient';

const TestComponent = component("TestComponent", ({test}) => {
    return (<h1>{test}</h1>);
})`;
        const output = fix(
            input,
            withRuleOptions({
                memoImport: "memo",
                memoModule: "react",
                passAreEqualToMemo: false,
            })
        );
        expect(output).toBeFixed();
        expect(output).toMatchSnapshot();
    });

    it("should convert component with raw props", () => {
        const input = `
import component from 'omniscient';

const TestComponent = component("TestComponent", (props) => {
return (<h1>{props.name}</h1>);
})`;
        const output = fix(input, withRuleOptions({ passAreEqualToMemo: false }));
        expect(output).toBeFixed();
        expect(output).toMatchSnapshot();
    });

    it("should convert component with destructed props", () => {
        const input = `
import component from 'omniscient';

const TestComponent = component("TestComponent", ({a, b}) => {
    return (<h1>{a}</h1>);
})`;
        const output = fix(input);
        expect(output).toBeFixed();
        expect(output).toMatchSnapshot();
    });

    it("should handle nameless components", () => {
        const input = `
import component from 'omniscient';

const TestComponent = component({
    getInitialState() {
        return { i: 1 };
    }
}, ({label}) => {
    return (<h1>{label}</h1>);
})`;
        const output = fix(input);
        expect(output).toBeFixed();
        expect(output).toMatchSnapshot();
    });
});
