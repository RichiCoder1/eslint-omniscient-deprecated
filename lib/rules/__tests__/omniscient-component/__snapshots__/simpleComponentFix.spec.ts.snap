// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple component conversions should convert component with destructed props 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';

class TestComponent extends Component {
    static displayName = \\"TestComponent\\";

    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    render() {
        const { a, b } = this.props;
        return <h1>{a}</h1>;
    }
}",
}
`;

exports[`simple component conversions should convert component with raw props 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';

class TestComponent extends Component {
    static displayName = \\"TestComponent\\";

    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    render() {
        const props = this.props;
        return <h1>{props.name}</h1>;
    }
}",
}
`;

exports[`simple component conversions should convert render only component 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';

class TestComponent extends Component {
    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    render() {
        return <h1>Test</h1>;
    }
}",
}
`;

exports[`simple component conversions should convert simple component to function when has memo 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { memo } from 'react';

import component from 'omniscient';

const TestComponent = memo(function TestComponent({ test }) {
    return <h1>{test}</h1>;
});
TestComponent.displayName = \\"TestComponent\\";",
}
`;

exports[`simple component conversions should convert simple component to function when has memo and pass areEqual 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { memo } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';

const TestComponent = memo(function TestComponent({ test }) {
    return <h1>{test}</h1>;
}, isEqualImmutable);
TestComponent.displayName = \\"TestComponent\\";",
}
`;

exports[`simple component conversions should convert simple pure component 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';

class TestComponent extends Component {
    static displayName = \\"TestComponent\\";

    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    render() {
        return <h1>Test</h1>;
    }
}",
}
`;

exports[`simple component conversions should convert simple pure component to function when has memo 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { memo } from 'react';

import component from 'omniscient';

const TestComponent = memo(function TestComponent() {
    return <h1>Test</h1>;
});
TestComponent.displayName = \\"TestComponent\\";",
}
`;

exports[`simple component conversions should handle nameless components 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';

class TestComponent extends Component {
    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    state = { i: 1 };

    render() {
        const { label } = this.props;
        return <h1>{label}</h1>;
    }
}",
}
`;
